scenes/test_array.txt

```diff
@@ -0,0 +1,243 @@
+*label entry
+*params module log_buffer
+*temp test_count 0
+*temp tests_passed 0
+
+
+*comment :::::: SET_ALL and CLEAN tests ::::::
+*gosub test_start "set all"
+*gosub_scene cslib_array set_all "test_arr" "JACKPOT"
+*gosub test_assert_equal "JACKPOT" test_arr_1
+*gosub test_assert_equal "JACKPOT" test_arr_2
+*gosub test_assert_equal "JACKPOT" test_arr_3
+*gosub test_assert_equal "JACKPOT" test_arr_4
+*gosub test_assert_equal "JACKPOT" test_arr_5
+*gosub test_assert_equal "JACKPOT" test_arr_6
+*gosub test_assert_equal "JACKPOT" test_arr_7
+*gosub test_assert_equal "JACKPOT" test_arr_8
+*gosub test_assert_equal "JACKPOT" test_arr_9
+*gosub test_assert (((test_arr_max = test_arr_count)) and (test_arr_max = 9)) "test_arr_max (${test_arr_max}) and test_arr_count (${test_arr_count}) should both be equal to 9"
```

Comment:
consider using multiple assert (instead of AND), so you know which one failed

Reply:
Because it uses a custom assert message this actually more obvious which one failed (the values are printed next to each), versus two generic "expected 9, received ?".